### ISSUE TITLE
reportCallInDefaultInitializer complains with Path as default parameter

### ISSUE THREAD

**Issue by FrancescElies** [microsoft/pyright#4673]
**Describe the bug**
Paths are [immutable](https://docs.python.org/3.7/library/pathlib.html#general-properties), hence it should be fine to have them as default parameters.

**To Reproduce**

`❯ cat pyright_path_in_signature.py`
```python
# pyright: reportCallInDefaultInitializer=true
from pathlib import Path


def myfun(a=Path(".")):
    ...
```

```
mytest on  master [?] via 🐍 v3.11.1
❯ pyright pyright_path_in_signature.py
No configuration file found.
No pyproject.toml file found.
stubPath C:\s\mytest\typings is not a valid directory.
Assuming Python platform Windows
Searching for source files
Found 1 source file
pyright 1.1.295
C:\s\mytest\pyright_path_in_signature.py
  C:\s\mytest\pyright_path_in_signature.py:5:13 - error: Function calls and mutable objects not allowed within parameter default value expression (reportCallInDefaultInitializer)
1 error, 0 warnings, 0 informations
Completed in 0.626sec
```

**Expected behavior**
If my understanding is correct it should not report an error on the example above.

Do I see that right?

**Comment by FrancescElies** [comment]
Sorry, I think a bit more about it and I'll reopen again and wait for an answer, I still think `Path(".")` should be allowed.

**Comment by FrancescElies** [comment]
Thanks for the feedback
